topydo/commands/AddCommand.py

```diff
@@ -34,14 +34,26 @@ def __init__(self, p_args, p_todolist,
             p_args, p_todolist, p_out, p_err, p_prompt)
         self.text = ' '.join(p_args)
         self.todo = None
+        self.from_file = False
 
-    def _preprocess_input_todo(self):
+    def _process_flags(self):
+        opts, args = self.getopt('f')
```

Comment:
I would have done this slightly differently: make `from_file` a `None` by default or the filename if given on the commandline with `-f`. So `getopt('f:')` and assign `self.from_file = value` below on line 44. Then you don't need to refer to `self.args[0]` on line 120, which may go wrong if the user neglects to pass on a filename. It's nicer to let `getopt` check whether the invocation is OK.